neo/core/spiketrain.py

```diff
@@ -641,6 +669,54 @@ def merge(self, other):
             train.lazy_shape = merged_lazy_shape
         return train
 
+    def _merge_array_annotations(self, other, sorting=None):
+        '''
+        Merges array annotations of 2 different objects.
+        The merge happens in such a way that the result fits the merged data
+        In general this means concatenating the arrays from the 2 objects.
+        If an annotation is only present in one of the objects, it will be omitted.
+        Apart from that the array_annotations need to be sorted according to the sorting of
+        the spikes.
+        :return Merged array_annotations
+        '''
+
+        assert sorting is not None, "The order of the merged spikes must be known"
+
+        # Make sure the user is notified for every object about which exact annotations are lost
+        warnings.simplefilter('always', UserWarning)
```

Comment:
Same as for DataObject: Should this filter setting be reset at the end of the method?